apps/confidential/examples_test.go

```diff
@@ -35,3 +36,53 @@ func ExampleNewCredFromCert_pem() {
 	}
 	fmt.Println(cred) // Simply here so cred is used, otherwise won't compile.
 }
+
+// ExampleAcquireTokenByCredential gives an example of acquiring token by credential.
+func ExampleAcquireTokenByCredential() {
+	//In this case, we are getting a credential using a secret.
+	// We could also use an assertion (NewCredFromAssertion) or a certificate (NewCredFromCert) to obtain a credential.
+	var secret = "the_secret"
+	cred, err := NewCredFromSecret(secret)
+	if err != nil {
+		log.Fatalf("ExampleAcquireTokenByCredential: acquring token by credential %v", err)
+	}
+	client, err := New("client_id", cred)
+	if err != nil {
+		log.Fatalf("ExampleAcquireTokenByCredential: acquring token by credential %v", err)
+	}
+
+	ctx := context.Background()
+	var tokenScope = []string{"the_scope"}
+
+	token, err := client.AcquireTokenByCredential(ctx, tokenScope)
+	if err != nil {
+		log.Fatalf("ExampleAcquireTokenByCredential: acquring token by credential %v", err)
+	}
+	fmt.Println(token)
+
+}
+
+// ExampleAcquireTokenByAuthCode gives an example of acquiring token by auth code.
+func ExampleAcquireTokenByAuthCode() {
+	//In this case, we are getting a credential using a secret.
+	// We could also use an assertion (NewCredFromAssertion) or a certificate (NewCredFromCert) to obtain a credential.
+	var secret = "the_secret"
+	cred, err := NewCredFromSecret(secret)
+	if err != nil {
+		log.Fatalf("ExampleAcquireTokenByAuthCode: acquring token by auth code %v", err)
+	}
+	client, err := New("client_id", cred)
+	if err != nil {
+		log.Fatalf("ExampleAcquireTokenByAuthCode: acquring token by auth code %v", err)
+	}
+
+	ctx := context.Background()
+	var tokenScope = []string{"the_scope"}
+
+	token, err := client.AcquireTokenByAuthCode(ctx, "auth_code", "redirect_uri", tokenScope)
```

Comment:
this should be named result as `result`  instead of `token`.
We return a token inside of [AuthResult](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/b1a1be475e73ac76ca97d2c399f8aa7b8fe0b986/apps/internal/base/base.go#L68)